SofaGLFW/src/SofaGLFW/SofaGLFWWindow.cpp

```diff
@@ -74,41 +76,44 @@ void SofaGLFWWindow::draw(sofa::simulation::NodeSPtr groot, sofa::core::visual::
     m_currentCamera->d_heightViewport.setValue(vparams->viewport()[3]);
 
     // matrices
-    double projectionMatrix[16];
-    double mvMatrix[16];
-    m_currentCamera->getOpenGLProjectionMatrix(projectionMatrix);
-    m_currentCamera->getOpenGLModelViewMatrix(mvMatrix);
+    double lastModelviewMatrix [16];
+    double lastProjectionMatrix [16];
+
+    vparams->getModelViewMatrix(lastModelviewMatrix);
+    vparams->getProjectionMatrix(lastProjectionMatrix);
+
+    m_currentCamera->getOpenGLProjectionMatrix(lastProjectionMatrix);
+    m_currentCamera->getOpenGLModelViewMatrix(lastModelviewMatrix);
 
     glViewport(0, 0, vparams->viewport()[2], vparams->viewport()[3]);
     glMatrixMode(GL_PROJECTION);
     glLoadIdentity();
-    glMultMatrixd(projectionMatrix);
+    glMultMatrixd(lastProjectionMatrix);
 
     glMatrixMode(GL_MODELVIEW);
     glLoadIdentity();
-    glMultMatrixd(mvMatrix);
+    glMultMatrixd(lastModelviewMatrix);
 
     // Update the visual params
     vparams->zNear() = m_currentCamera->getZNear();
     vparams->zFar() = m_currentCamera->getZFar();
-    vparams->setProjectionMatrix(projectionMatrix);
-    vparams->setModelViewMatrix(mvMatrix);
-
-    sofa::simulation::node::draw(vparams, groot.get());
+    vparams->setProjectionMatrix(lastProjectionMatrix);
+    vparams->setModelViewMatrix(lastModelviewMatrix);
 
+    simulation::node::draw(vparams, groot.get());
```

Comment:
I don't think those changes are necessary because you get the model view matrix twice. In the end, it's the second call (`getOpenGLModelViewMatrix`) which overwrites the first, so the first call is useless.